_src/doc-utils.ts

```diff
@@ -0,0 +1,348 @@
+    );
+  }
+};
+
+/**
+ * Call this function to make the wallet switch to the desired chain
+ * @param chainId designed chain in HexString
+ * @param ethereum ethereum inpage provider (e.g.: provider loaded by Metamask)
+ */
+const switchToChain = async (chainId: string, ethereum: MetaMaskInpageProvider) => {
+  if (!isChainIdFormatValid(chainId)) {
+    throw new Error(`chainId '${chainId}' must be hexString`);
+  }
+  validateEthereumApi(ethereum);
+  await ethereum.request({
+    method: 'wallet_switchEthereumChain',
+    params: [{ chainId: chainId }],
+  });
+  console.log(`Succesfully switched to chain ${chainId} in metamask`);
+};
+
+/**
+ * Call this function to add a new chain to the wallet. Should only be called if the chain doesn't exist already in the wallet
+ * @param chainId designed chain in HexString
+ * @param ethereum ethereum in page provider (e.g.: provider loaded by Metamask)
+ */
+const addChainToWallet = async (chainId: string, ethereum: MetaMaskInpageProvider) => {
+  if (!isChainIdFormatValid(chainId)) {
+    throw new Error(`chainId '${chainId}' must be hexString`);
+  }
+  validateEthereumApi(ethereum);
+  const chains = await (await fetch(chainsReferenceData)).json();
+  const chain = chains.find((c: any) => toHex(c.chainId) === chainId);
+  if (!chain || !chain.chainId) {
+    console.log(chain);
+    throw new Error(`Chain with chainId '${chainId}' not found in reference data`);
+  }
+
+  const params: AddEthereumChainParameter = {
+    chainId: chainId,
+    chainName: chain.name,
+    nativeCurrency: chain?.nativeCurrency,
+    blockExplorerUrls:
+      chain.explorers && chain.explorers.length > 0 && chain.explorers[0].url
+        ? chain.explorers.map((explorer: any) => explorer.url)
+        : [chain.infoURL],
+    rpcUrls: chain.rpc,
+  };
+
+  const [signerAddress] = (await ethereum.request({ method: 'eth_requestAccounts' })) as string[];
+  await ethereum.request({
+    method: 'wallet_addEthereumChain',
+    params: [params, signerAddress],
+  });
+  console.log(`Chains ${chainId} of params ${JSON.stringify(params)} succesfully added to wallet`);
+};
+
+/**
+ * Functions which validates the format of Link address and interfacs with the contract to make sure
+ * its metadata (e.g.: symbol) is valid
+ * @param address
+ * @param provider
+ */
+const validateLinkAddress = async (address: string, provider: Web3Provider) => {
+  if (!isAddressFormatValid(address)) {
+    throw new Error(`Something went wrong. format of address '${address}' not correct`);
+  }
+
+  // The Contract object
+  const linkContract = new ethers.Contract(address, LinkToken, provider);
+  let name: string, symbol: string, decimals: BigNumberish;
+  try {
+    name = await linkContract.name();
+    symbol = await linkContract.symbol();
+    decimals = await linkContract.decimals();
+  } catch (error) {
+    throw new Error(`Error occured while trying to fetch linkContract metadata  ${error}`);
+  }
+  /*
```

Comment:
There should not be commented out checks in production code.

Reply:
ok @obelisk will remove this code in the next push

```
/*
  if (!equalOrContain(linkToken.name, name)) {
    // found inconsistencies on some networks . E.g: xdai, HECO chain..Etc
    throw new Error(`Error linkContract name. '${linkToken.name}' !== '${name}'`);
  }
  */
```